modules/reitit-core/src/reitit/coercion.cljc

```diff
@@ -83,42 +119,66 @@
                 format (extract-request-format request)
                 result (coercer value format)]
             (if (error? result)
-              (request-coercion-failed! result coercion value in request)
-              result)))))))
+              (if coerce-all-on-error?
+                [nil {:error result
+                      :in (if (= in :body-params) :body in)
+                      :coercion coercion
+                      :value value}]
+                (request-coercion-failed-fail-fast! result coercion value in request))
+              [result])))))))
 
 (defn extract-response-format-default [request _]
   (-> request :muuntaja/response :format))
 
-(defn response-coercer [coercion body {:keys [extract-response-format]
-                                       :or {extract-response-format extract-response-format-default}}]
+(defn response-coercer [coercion body {::keys [extract-response-format coerce-all-on-error?]
+                                       :or {extract-response-format extract-response-format-default
+                                            coerce-all-on-error? false}}]
   (if coercion
     (if-let [coercer (-response-coercer coercion body)]
       (fn [request response]
         (let [format (extract-response-format request response)
               value (:body response)
               result (coercer value format)]
           (if (error? result)
-            (response-coercion-failed! result coercion value request response)
-            result))))))
+            (if coerce-all-on-error?
+              [nil {:error result
+                    :in :body
+                    :coercion coercion
+                    :value value}]
+              (response-coercion-failed-fail-fast! result coercion value request response))
+            [result]))))))
+
+(defn encode-single-error [data]
+  (-encode-error (:coercion data) (update data :coercion -get-name)))
 
 (defn encode-error [data]
-  (-> data
-      (dissoc :request :response)
-      (update :coercion -get-name)
-      (->> (-encode-error (:coercion data)))))
+  (if (not-any? #(contains? data %)
+                [:body :form-params :query-params :headers :path-params])
```

Comment:
This might not be the cleanest way to tell the old and new format apart. Perhaps it'd been better to handle this via an additional key to the `ex-info` data?